crates/paralegal-spdg/src/lib.rs

```diff
@@ -459,6 +459,18 @@ define_index_type! {
     pub struct CallSiteOrDataSinkIndex for CallSiteOrDataSink = u32;
 }
 
+impl From<&CallSite> for CallSiteOrDataSink {
```

Comment:
I think these should be `From<T>` rather than `From<&T>`. It is bad form to hide a `.clone()` in a `from`. Also the way it is now you couldn't use `.into()` on a newly constructed `CallSite` or `DataSink`.

Reply:
This might also be a good time to add some `as_*` methods, e.g.

```rs
impl CallSiteOrDataSink {
    pub fn as_call_site(&self) -> Option<&CallSite> {
        match self {
            Self::CallSite(cs) => Some(cs),
            _ => None
        }
    }

    pub fn as_data_sink(&self) -> Option<&DataSink> {
        match self {
            Self::DataSink(ds) => Some(ds),
            _ => None,
        }
    }
}
```

This is particularly helpful with something like `Iterator::filter_map`

Reply:
done